java_tests/src/test/java/org/lsst/sal/BaseTestCase.java

```diff
@@ -0,0 +1,75 @@
+package org.lsst.sal;
+
+import org.junit.Assert;
+import org.junit.Before;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Random;
+
+
+public class BaseTestCase {
+
+    // Standard sleep time in ms.
+    final long STD_SLEEP = 1L;
+    SAL_Test remote;
+    SAL_Test controller;
+
+    // Depth of DDS read queues (which is also the depth of the write queues).
+    // This must be at least as long as the actual depth for some tests to pass.
+    // If it ever becomes possible to read or write the depth in SALPY, update
+    // these tests to do that, to eliminate the dependence on the default depth.
+    final int READ_QUEUE_DEPTH = 100;
+
+    @Before
+    public void setUp() {
+        Random rand = new Random(2000L);
```

Comment:
Please generate a random LSST_DDS_DOMAIN_PREFIX env var before constructing the remote and controller. That is potentially useful for keeping tests from colliding with each other. I think it won't be a problem that the JVM caches information, because SAL reads the env var when it creates the `SAL_<name>` object.

Reply:
Would it be OK if I addressed this as well in DM-32914?